src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2987,14 +2987,28 @@ public static bool IsEnumTypeMemberAccessContext(this SyntaxTree syntaxTree, Sem
                 .FindTokenOnLeftOfPosition(position, cancellationToken)
                 .GetPreviousTokenIfTouchingWord(position);
 
-            if (!token.IsKind(SyntaxKind.DotToken) ||
-                !token.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+            if (!token.IsKind(SyntaxKind.DotToken))
+            {
+                return false;
+            }
+
+            SymbolInfo leftHandBinding;
+            if (token.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression))
+            {
+                var memberAccess = (MemberAccessExpressionSyntax)token.Parent;
+                leftHandBinding = semanticModel.GetSymbolInfo(memberAccess.Expression, cancellationToken);
+            }
+            else if (token.Parent.IsKind(SyntaxKind.QualifiedName, out QualifiedNameSyntax? qualifiedName) &&
+                token.Parent.IsParentKind(SyntaxKind.IsExpression))
```

Comment:
it might be a good idea to validate that the qualifiedName is the RHS of hte is-expression, not the left.